src/Microsoft.SqlTools.ServiceLayer/DacFx/Contracts/GenerateTSqlModelRequest.cs

```diff
@@ -0,0 +1,53 @@
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+using Microsoft.SqlTools.Hosting.Protocol.Contracts;
+using Microsoft.SqlTools.ServiceLayer.Utility;
+using Microsoft.SqlServer.Dac.Model;
+
+namespace Microsoft.SqlTools.ServiceLayer.DacFx.Contracts
+{
+    /// <summary>
+    /// Parameters to generate a SQL model
+    /// </summary>
+    public class GenerateTSqlModelParams
+    {
+        /// <summary>
+        /// Project uri
+        /// </summary>
+        public string ProjectUri { get; set; }
+
+        /// <summary>
+        /// The version of Sql Server to target
+        /// </summary>
+        public string ModelTargetVersion { get; set; }
+
+        /// <summary>
+        /// Gets or sets the Sql script file paths.
+        /// </summary>
+        public string[]? FilePaths { get; set; }
```

Comment:
Should this be optional? We would expect an array of paths regardless, right? Even if it's just an empty one (although I'm not sure what that would accomplish, although I suppose starting with a "blank" model could be useful for something)